pkgs/by-name/au/authentik/package.nix

```
@@ -0,0 +1,236 @@
+    '';
+
+    nativeBuildInputs = [ python3.pkgs.poetry-core ];
+    propagatedBuildInputs = with python3.pkgs; [
+      argon2-cffi
+      celery
+      channels
+      channels-redis
+      colorama
+      dacite
+      daphne
+      deepmerge
+      defusedxml
+      django
+      django-filter
+      django-guardian
+      django-model-utils
+      django-prometheus
+      django-redis
+      djangorestframework
+      djangorestframework-guardian2
+      docker
+      drf-spectacular
+      duo-client
+      facebook-sdk
+      flower
+      geoip2
+      gunicorn
+      httptools
+      kubernetes
+      ldap3
+      lxml
+      opencontainers
+      packaging
+      paramiko
+      psycopg
+      pycryptodome
+      pydantic
+      pydantic-scim
+      pyjwt
+      pyyaml
+      requests-oauthlib
+      sentry-sdk
+      structlog
+      swagger-spec-validator
+      twilio
+      twisted
+      ua-parser
+      urllib3
+      uvicorn
+      uvloop
+      watchdog
+      webauthn
+      websockets
+      wsproto
+      xmlsec
+      zxcvbn
+      jsonpatch
+    ] ++ [
+      codespell
+    ];
+
+    postInstall = ''
+      mkdir -p $out/web $out/website
+      cp -r lifecycle manage.py $out/${python3.sitePackages}/
+      cp -r blueprints $out/
+      cp -r ${webui}/dist ${webui}/authentik $out/web/
+      cp -r ${website} $out/website/help
+    '';
+  };
+
+  proxy = buildGoModule {
+    pname = "authentik-proxy";
+    inherit version;
+
+    src = runCommand "authentik-proxy-source" {} ''
+      mkdir -p $out/web
+      mkdir -p $out/authentik
+
+      cp -r ${src}/cmd ${src}/internal $out/
+      cp -r ${src}/authentik/lib $out/authentik/
+      cp ${src}/web/static.go $out/web/
+
+      cp ${src}/web/robots.txt ${src}/web/security.txt $out/web/
+      cp ${src}/go.mod ${src}/go.sum $out/
+    '';
+
+    patchPhase = ''
+      substituteInPlace internal/gounicorn/gounicorn.go \
+        --replace './lifecycle' "${authentik-django}/${python3.sitePackages}/lifecycle"
+      substituteInPlace web/static.go \
+        --replace './web' "${authentik-django}/web"
+      substituteInPlace internal/web/static.go \
+        --replace './web' "${authentik-django}/web"
+    '';
+
+    CGO_ENABLED = 0;
+
+    vendorHash = "sha256-8F9emmQmbe7R+xtGrjV5ht0adGasU6WAvLa8Wxr+j8M=";
+
+    postInstall = ''
+      mv $out/bin/server $out/bin/authentik
+    '';
+
+    subPackages = [ "cmd/server" ];
+  };
+
+in stdenvNoCC.mkDerivation {
+  pname = "authentik";
+  inherit src version;
+
+  patchPhase = ''
+    rm Makefile
+    patchShebangs lifecycle/ak
+
+    # This causes issues in systemd services
+    substituteInPlace lifecycle/ak \
+      --replace 'printf' '>&2 printf' \
+      --replace '> /dev/stderr' ""
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp -r lifecycle/* $out/bin/
+
+    wrapProgram $out/bin/ak --prefix PATH : ${
+      lib.makeBinPath [ (python3.withPackages (_: [authentik-django])) proxy ]
+    } --set TMPDIR /dev/shm \
```

```suggestion
    wrapProgram $out/bin/ak \
      --prefix PATH : ${lib.makeBinPath [ (python3.withPackages (ps: [ps.authentik-django])) proxy ]} \
      --set TMPDIR /dev/shm \
```
we don't want to accidentally a wrong python interpreter. Also setting TMPDIR to /dev/shm seems strange.

---

The TMPDIR is taken from upstream

---

Also the package is the one that is build earlier, it is not part of the python package set. And it cannot be out of sync because it is built from the same python as this

---

how would I do that?

---

I had something like this https://github.com/NixOS/nixpkgs/blob/master/pkgs/servers/web-apps/healthchecks/default.nix#L8-L12 in mind